include/sys/arc_impl.h

```diff
@@ -237,6 +270,11 @@ struct arc_buf_hdr {
 	l2arc_buf_hdr_t		b_l2hdr;
 	/* L1ARC fields. Undefined when in l2arc_only state */
 	l1arc_buf_hdr_t		b_l1hdr;
+	/*
+	 * Encryption parameters. Defined only when ARC_FLAG_ENCRYPTED
+	 * is set and the L1 header exists.
+	 */
+	arc_buf_hdr_crypt_t b_crypt_hdr;
```

Comment:
One of my big goals for this patch was to avoid impacting performance of unencrypted datasets at all if possible. Its a little more complicated like this, but the code is already written and I'd feel bad making a change that hurts performance (even if  it is in a very minor way). Let me know if you feel strongly about it either way.